geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/key/AbstractRenameIntegrationTest.java

```diff
@@ -188,38 +245,39 @@ public void shouldNotDeadlock_concurrentRenames_givenStripeContention()
   }
 
   @Test
-  public void shouldThrowError_givenKeyDeletedDuringRename()
-      throws ExecutionException, InterruptedException {
-    CyclicBarrier startCyclicBarrier = new CyclicBarrier(2);
-    ExecutorService pool = Executors.newFixedThreadPool(2);
-
-    for (int i = 0; i < 100; i++) {
-      jedis.set("{user1}oldKey", "foo");
+  public void shouldError_givenKeyDeletedDuringRename() {
+    int iterations = 2000;
 
-      Runnable renameOldKeyToNewKey = () -> {
-        cyclicBarrierAwait(startCyclicBarrier);
+    final AtomicReference<RuntimeException> renameException = new AtomicReference<>(null);
 
-        jedis.rename("{user1}oldKey", "{user1}newKey");
-      };
+    jedis.set("{user1}oldKey", "foo");
 
-      Runnable deleteOldKey = () -> {
-        cyclicBarrierAwait(startCyclicBarrier);
-        jedis.del("{user1}oldKey");
-      };
-
-      Future<?> future1 = pool.submit(renameOldKeyToNewKey);
-      Future<?> future2 = pool.submit(deleteOldKey);
-
-      try {
-        future1.get();
-        assertThat(jedis.get("{user1}newKey")).isEqualTo("foo");
-      } catch (Exception e) {
-        assertThat(e).hasMessageContaining("no such key");
-      }
-      future2.get();
-
-      assertThat(jedis.get("{user1}oldKey")).isNull();
+    try {
+      new ConcurrentLoopingThreads(iterations,
```

Comment:
This test still ends up only running a single iteration since an exception causes it to end. That wasn't the intent of the original test.

Reply:
I changed the semantics of the test to do this because of the suggestion from Donal here:
https://github.com/apache/geode/pull/6877#discussion_r714212553

Reply:
His interpretation of the original test was that the point was to get the error to be thrown and that's it so the test can exit with success as soon as that exception is thrown.   The test does not necessarily only run a single iteration because the rename can occur before the delete in which case the next iteration is executed.

Reply:
I accepted his interpretation of that and went with it

Reply:
OK, I guess that makes sense - especially given the name of the test method too.